webnote.py

```diff
@@ -50,18 +62,238 @@
     response = ''
 
     response += 'HTTP/1.1 200 OK\r\n'
-
+    
     try:
+        #print 'userloged = ', userloged
         with open('.notes') as note_file:
             note_dict = json.loads(note_file.read())
     except IOError:
         note_dict = dict()
+
+    note_list = []
+    for key in note_dict:
+            note_list.append(note_dict[key])
+    
+    try:
+        con = lite.connect('datas.db')
+        with con:
+            cur = con.cursor()
+            cur.execute("CREATE TABLE Notes(Id INTEGER PRIMARY KEY, Key TEXT UNIQUE NOT NULL, Note TEXT NOT NULL)")
+    except lite.OperationalError:
+        pass
+   
+
+    try:
+        con1 = lite.connect('datas.db')
+        with con1:
+            cur1 = con1.cursor()
+            cur1.execute("CREATE TABLE Users(Id INTEGER PRIMARY KEY, Name TEXT UNIQUE NOT NULL, Password TEXT NOT NULL)")
+    except lite.OperationalError:
+        pass
+
+    print loged
+    print userloged
+    print resource    
+
+    if resource == '/auth':
+        response += 'Content-Type: text/html; charset=utf-8\r\n'
+        response += '\r\n'
+        if method == 'GET':
+            response += """
+            <html>
+                <head>
+                    <meta http-equiv="content-type" content="text/html; charset=utf-8"/>
+                </head>
+                <body>
+                <form action="/auth" method="POST">
+                    <h2>Insira seu usuario e senha!</h2>
+                    User: <input type="text" name="user"/>
+                    <br/>
+                    Password: <input type="text" name="password"/>
+                    <br/>
+                    <input type="submit" value="Add"/>
+                </form>
+                </body>
+            </html>
+            """
+        elif method == 'POST':
+            print data
+            new_dict = parse_qs(data)
```

Comment:
Seria esse caso aqui?

```
if len(data) != int (request_headers['content-length']):
data += conn.recv(1024)
else:
break
```

Acho que to esquecendo também que em uma requisição aleatória poderiam ir dados que o servidor não deveria suportar, como falamos antes. Nesse caso deveria repetir o trecho seguinte?

```
if request_headers['content-type'] != 'application/x-www-form-urlencoded':
response = ''
response += 'HTTP/1.1 501 Not Implemeted\r\n'
response += 'Content-Type: text/html; charset=utf-8\r\n'
response += '\r\n'
response += """
<html>
<head>
<meta http-equiv="content-type" content="text/html; charset=utf-8"/>
</head>
<body>
<h1>ERROR 501<h1>
</body>
</html>
"""
data = '1'
break
```